core/src/main/scala/org/apache/spark/util/random/RandomSampler.scala

```diff
@@ -107,21 +114,21 @@ class BernoulliCellSampler[T](lb: Double, ub: Double, complement: Boolean = fals
 
   override def setSeed(seed: Long): Unit = rng.setSeed(seed)
 
+  override def sample(): Int = {
+    if (ub - lb <= 0.0) {
+      if (complement) 1 else 0
+    } else {
+      val x = rng.nextDouble()
+      val n = if ((x >= lb) && (x < ub)) 1 else 0
+      if (complement) 1 - n else n
+    }
+  }
+
   override def sample(items: Iterator[T]): Iterator[T] = {
     if (ub - lb <= 0.0) {
       if (complement) items else Iterator.empty
```

Comment:
I think we don't need to optimize this corner case, it's fine to use the default `sample`